File: datahub-dao/src/main/java/com/linkedin/datahub/models/view/DatasetView.java
package com.linkedin.datahub.models.view;

import java.util.List;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@NoArgsConstructor
public class DatasetView {

  private String platform;

  private String nativeName;

  private String fabric;

  private String uri;

  private String description;

  private String nativeType;

  private String properties;

  private List<String> tags;

  private Boolean removed;

  private Boolean deprecated;

  private String deprecationNote;

  private Long decommissionTime;

  private Long createdTime;

  private Long modifiedTime;
}

File: datahub-web/packages/data-portal/mirage/factories/dataset-view.ts
import { Factory, faker } from 'ember-cli-mirage';
import { DatasetPlatform } from '@datahub/metadata-types/constants/entity/dataset/platform';

export default Factory.extend({
  createdTime: faker.date.past(2),
  deprecated: true,
  deprecationNote: faker.lorem.words(5),
  description: faker.lorem.words(7),
  fabric: null,
  modifiedTime: faker.date.recent(),
  nativeType: null,
  platform: faker.list.random(...Object.values(DatasetPlatform)),
  properties: '{}',
  removed: faker.random.boolean(),
  tags: null,
  bucket: null,
  // in search there is a name field
  name(i: number) {
    return `dataset-${i}`;
  },
  nativeName() {
    return this.name;
  },
  uri() {
    const { platform, name, bucket } = this;
    const datasetPath = bucket ? `${bucket}.${name}` : name;
    return `urn:li:dataset:(urn:li:dataPlatform:${platform},${datasetPath},PROD)`;
  },
  urn() {
    return this.uri;
  }
});
